core/src/main/java/nl/nn/adapterframework/stream/Message.java

```diff
@@ -101,11 +100,11 @@ private Message(Map<String, Object> context, Object request, Class<?> requestCla
 		this.requestClass = requestClass != null ? ClassUtils.nameOf(requestClass) : ClassUtils.nameOf(request);
 	}
 
-	private Message(Map<String, Object> context, Object request) {
+	private Message(MessageContext context, Object request) {
 		this(context, request, request != null ? request.getClass() : null);
 	}
 
-	public Message(String request, Map<String, Object> context) {
+	public Message(String request, MessageContext context) {
```

Comment:
Can we add `@Nonnullable` annotations to all these constructors as well, for the context at least?
(Perhaps also for the request, although there are some private constructors that do allow NULL request).

I would like at least context to never be NULL though and that is not guaranteed now.